clients/fides-js/src/lib/consent-types.ts

```diff
@@ -700,10 +711,14 @@ export type RecordConsentServedRequest = {
   tcf_special_features?: Array<number>;
   tcf_system_consents?: Array<string>;
   tcf_system_legitimate_interests?: Array<string>;
+  /**
+   * @deprecated has no effect; use privacy_experience_config_history_id instead!
+   */
   privacy_experience_id?: string;
+  privacy_experience_config_history_id?: string;
   user_geography?: string;
   acknowledge_mode?: boolean;
-  serving_component: ServingComponent;
+  serving_component: string; // NOTE: uses a generic string instead of an enum
```

Comment:
Why?

Reply:
Fun fact: Typescript enums aren't actually strings. They compile to Javascript objects at runtime which is surprising and therefore can't be used directly when serializing to an API call, etc. I don't like that at all... we'd be better served with union types for these!

Hard to explain succinctly but happily would dogpile with anyone, I definitely went into the Typescript docs at one point when playing with these recently: https://www.typescriptlang.org/docs/handbook/enums.html